exporter/azuremonitorexporter/traceexporter.go

```diff
@@ -653,6 +658,39 @@ func (exporter *traceExporter) pushTraceData(
 
 	for _, wireFormatSpan := range traceData.Spans {
 		if envelope, err := exporter.spanToEnvelope(exporter.config.InstrumentationKey, wireFormatSpan); err == nil && exporter.transportChannel != nil {
+
+			// Attach node level attributes to envelope and data as appropriate
+			if traceData.Node != nil && traceData.Node.Attributes != nil {
+				// Augment the envelope and envelope data with node level attributes
+				// Configure the ai.cloud.role and ai.cloud.roleinstance on the envelope tags itself, then copy the node attributes to the envelope data as well.
+				// Assumes the node level attribute key/values correspond to:
+				// https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/data-resource-semantic-conventions.md
+				if serviceName, ok := traceData.Node.Attributes[resourceAttributeKeyServiceName]; ok {
+					envelope.Tags[contracts.CloudRole] = serviceName
```

Comment:
perhaps also prepend `service.namespace` to is.

Reply:
good idea.